draft-ietf-quic-http.md

```diff
@@ -585,6 +585,116 @@ The payload consists of:
+connection, although a new connection can be established for new streams.
+
+If the receiver of the GOAWAY has sent requests on streams with a higher stream
+identifier than what is indicated in the GOAWAY frame, those streams are not or
+will not be processed.  The receiver of the GOAWAY frame can treat the requests
+as though they had never been sent at all, thereby allowing those requests to be
+retried later on a new connection.  Servers SHOULD treat requests on streams
+above this ID as stream errors of type HTTP_REQUEST_CANCELLED.
+
+Endpoints SHOULD send a GOAWAY frame when the closing of a connection is known
+in advance, even if the window is small, so that the remote peer can know
+whether a stream has been partially processed or not.  For example, if an HTTP
+client sends a POST at the same time that a server closes a QUIC connection, the
+client cannot know if the server started to process that POST request if the
+server does not send a GOAWAY frame to indicate what streams it might have acted
+on.
+
+For unexpected closures caused by error conditions, transport-level mechanisms
+SHOULD be used.  However, a GOAWAY MAY still be sent to provide additional
+detail to clients.
+
+~~~~~~~~~~  drawing
+ +---------------------------------------------------------------+
+ |                   Last-Stream-ID (32)                         |
+ +---------------------------------------------------------------+
+ |                     [Error Code (32)]                         |
+ +---------------------------------------------------------------+
+ |                 [Additional Debug Data (*)]                 ...
+ +---------------------------------------------------------------+
+~~~~~~~~~~
+ {: #fig-goaway title="GOAWAY frame payload"}
+
+The GOAWAY frame does not define any flags.
+
+The GOAWAY frame applies to the connection, not a specific stream. An endpoint
+MUST treat a GOAWAY frame on a stream other than the control stream as a
+connection error ({{errors}}) of type HTTP_WRONG_STREAM.
+
+The last stream identifier in the GOAWAY frame contains the highest-numbered
+stream identifier for which the sender of the GOAWAY frame might have taken some
+action on or might yet take action on.  All streams up to and including the
+identified stream might have been processed in some way.  The last stream
+identifier can be set to 0 if no streams were processed.
+
+> Note: In this context, "processed" means that some data from the stream was
+> passed to some higher layer of software that might have taken some action as a
+> result.
+
+If a connection terminates without a GOAWAY frame, the last stream identifier is
+effectively the highest possible stream identifier.
+
+Reattempting requests on streams with lower- or equal-numbered identifiers is
+not possible, with the exception of idempotent actions like HTTP GET, PUT, or
+DELETE.  Any protocol activity that used higher-numbered streams can be safely
```

Comment:
This is probably cribbed from RFC 7450 (as is most of this), however it could be improved.  I think that you want to concentrate first on what this guarantees (requests on higher stream numbers can be retried automatically), then note that other streams cannot be automatically retried unless it is otherwise permitted (for example, a request with an idempotent method).